pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -62,6 +76,48 @@ def visit_call(self, node):
         if isinstance(node.parent, (astroid.For, astroid.Comprehension)):
             self.add_message("consider-iterating-dictionary", node=node)
 
+    def _check_consider_using_maxsplit_arg(self, node: astroid.Call) -> None:
+        """Add message when accessing first or last elements of a str.split() or str.rsplit(),
+        or when split/rsplit with max_split=1 is used"""
+
+        # Check if call is split() or rsplit()
+        if (
+            isinstance(node.func, astroid.Attribute)
+            and node.func.attrname in ("split", "rsplit")
+            and isinstance(utils.safe_infer(node.func), astroid.BoundMethod)
+        ):
+            try:
+                sep = utils.get_argument_from_call(node, 0, "sep").value
+            except utils.NoSuchArgumentError:
+                return
+
+            # Check if maxsplit is set, and ignore checking if maxsplit is > 1
+            try:
+                maxsplit = utils.get_argument_from_call(node, 1, "maxsplit").value
+                if maxsplit > 0:
+                    return
+            except utils.NoSuchArgumentError:
+                maxsplit = 0
```

Comment:
Do we still need to store `maxsplit`? And for the `except` case: shouldn't there be a `return`?

Reply:
```python
            try:
                # Ignore if maxsplit arg has been set
                _ = utils.get_argument_from_call(node, 1, "maxsplit").value
                return
            except utils.NoSuchArgumentError:
                pass
```
I changed it to this, and added comment for readability